src/js/view/view.js

```diff
@@ -739,7 +744,13 @@ function View(_api, _model) {
             case STATE_IDLE:
             case STATE_ERROR:
             case STATE_COMPLETE:
-                _model.change('playlistItem', setPosterImage);                
+                // Set the poster image for videos before playback starts (idle), when the playlist ends (complete),
+                // or when an error is encountered. We don't get to the idle state between playlist items because of RAF
+
+                if (_model.mediaModel.get('mediaType') === 'video') {
+                    _model.change('playlistItem', setPosterImage);
```

Comment:
Why don't we just make this `setPosterImage(_model);`? The "change" listener will result in the poster being set on on the next playlist item regardless of state.